targets/unity-v2/source/ExampleTestProject/Assets/PlayFabEditorExtensions/Editor/Scripts/Components/TitleDataViewer.cs

```diff
@@ -14,21 +14,113 @@ public class TitleDataViewer : UnityEditor.Editor
         public static TitleDataEditor tdEditor;
         public Vector2 scrollPos = Vector2.zero;
         private bool showSave = false;
+        private static int focusIndex;
+        private static bool isShiftKeyPressed = false;
 
+
+        private static void shiftKeyHandler()
+        {
+            var e = Event.current;
+            if (e.keyCode == KeyCode.LeftShift || e.keyCode == KeyCode.RightShift)
+            {
+                isShiftKeyPressed = true;
+            }
+
+            if (e.type == EventType.KeyUp && (e.keyCode == KeyCode.LeftShift || e.keyCode == KeyCode.RightShift))
+            {
+                isShiftKeyPressed = false;
+            }
+        }
+        private static void titleHandler()
+        {
+            var e = Event.current;
+            shiftKeyHandler();
+            if (e.type == EventType.KeyUp && e.keyCode == KeyCode.Tab)
+            {
+                if (!isShiftKeyPressed)
+                {
+                    switch (focusIndex)
+                    {
+                        case 0:
+                            EditorGUI.FocusTextInControl("refresh");
+                            focusIndex = 1;
+                            break;
+                        case 1:
+                            EditorGUI.FocusTextInControl("add");
+                            focusIndex = 2;
+                            break;
+                        case 2:
+                            EditorGUI.FocusTextInControl("new_value_text_field");
+                            focusIndex = 3;
+                            break;
+                        case 3:
+                            EditorGUI.FocusTextInControl("edit");
+                            focusIndex = 4;
+                            break;
+                        case 4:
+                            EditorGUI.FocusTextInControl("clear");
+                            focusIndex = 5;
+                            break;
+                        case 5:
+                            EditorGUI.FocusTextInControl("save");
+                            focusIndex = 6;
+                            break;
+                        case 6:
+                            EditorGUI.FocusTextInControl("refresh");
+                            focusIndex = 0;
+                            break;
+
+                    }
+                }
+                else
+                {
+                    switch (focusIndex)
```

Comment:
I expected this switch block to basically be the opposite of the one above. Is there a reason it has fewer cases? Why is "new_value_text_field" missing?

Reply:
I think there might be other index numbering issues here as well.

Reply:
Missing code snipped has been added.